spec/message.abnf

```diff
@@ -45,12 +45,8 @@ quoted-char = %x0-5B         ; omit \
             / %x7D-D7FF      ; omit surrogates
             / %xE000-10FFFF
 
-; based on https://www.w3.org/TR/xml/#NT-Nmtoken,
-; but cannot start with U+002D HYPHEN-MINUS or U+003A COLON ":"
-unquoted = unquoted-start *name-char
+unquoted       = unquoted-start *name-char
 unquoted-start = name-start / DIGIT / "."
-               / %xB7 / %x300-36F / %x203F-2040
```

Comment:
Hang on, what happened here? Why are we further restricting `unquoted-start` from what it was before? On the call we talked about _expanding_ `unquoted` to something like:

```abnf
unquoted = unquoted-start *unquoted-char
unquoted-char = name-char / ":"
```

There is no reason to leave out any of these characters from `unquoted-start`, and if there is, that's beyond the scope of this PR.

Reply:
Unquoted cannot start with `:` or it would be a function identifier. The character we want to add is `-`.

The bottom part is my mistake.

Reply:
One other note: adding `-` would make `unquoted-start` identical to `name-char`, so `unquoted` becomes just `*name-char`

Reply:
> Unquoted cannot start with `:` or it would be a function identifier. The character we want to add is `-`.

Agreed, `unquoted` cannot start with `:`. But the `unquoted-char` that I propose introducing is for characters after the first, which before this PR _could_ include `:` because it was a permitted `name-char` character. As it no longer is, `unquoted-char` adds it back in so that this PR does not change anything about `unquoted`.

Adding `-` to `unquoted-start` is beyond the scope of this PR, and depends on resolving open-close. At earliest it should be done in the PR implementing open-close in the spec.

> The bottom part is my mistake.

Ah! Thank you for fixing it already.

Reply:
I think we should leave leading `-` in the other ticket and just ensure here that unquoted literals can start with decimal digits while names cannot.